packages/odyssey-react/src/components/Tab/index.tsx

```diff
@@ -0,0 +1,176 @@
+
+  /**
+   * The tab id attribute. Automatically generated if not provided.
+   */
+  id: string,
+
+  /**
+   * An aria label to be placed on the tab component. It
+   * should describe the purpose of the tabs.
+   */
+  ariaLabel: string,
+
+  /**
+   * Specifes which Tabs.Panel should be selected on mount
+   */
+  selectedId?: string,
+  
+  /**
+   * Callback when the selected tab is changed.
+   */
+  onTabChange?: (newId: string) => void
+}
+
+export type PropsTabsContainer = { 
+  children: ReactElement | ReactElement[],
+  id: string,
+  ariaLabel: string
+}
+
+export type PropsTabsPanelContainer = { 
+  children: ReactElement | ReactElement[]
+}
+
+export type PropsTabsPanel = { 
+  children: ReactText | ReactElement | ReactElement[],
+  label: string,
+  id?: string,
+  selected?: boolean
+}
+
+export type PropsTabsList = { 
+  children: ReactElement | ReactElement[],
+}
+
+export type PropsTab = { 
+  children: ReactText,
+  id: string,
+  ariaControls: string,
+  selected?: boolean,
+  onClick: () => void
+}
+
+export type StaticComponents = {
+  Container: FunctionComponent<PropsTabsContainer>,
+  PanelContainer: FunctionComponent<PropsTabsPanelContainer>,
+  Panel: FunctionComponent<PropsTabsPanel>,
+  List: FunctionComponent<PropsTabsList>,
+  Tab: FunctionComponent<PropsTab>
+}
+
+/**
+ * Navigation component used to organize content by grouping similar information on the 
+ * same page. They allow content to be viewed without having to navigate away from that page 
+ * or route.
+ * 
+ * @component
+ * @example
+ * <Tabs id="example-tabs" selectedId="tab-2" ariaLabel="Describes the purpose of the tabs.">
+ *   <Tabs.Panel id="tab-1" label="Tab One">TabPanel One Content</Tabs.Panel>
+ *   <Tabs.Panel id="tab-2" label="Tab Two">TabPanel Two Content</Tabs.Panel>
+ *   <Tabs.Panel id="tab-3" label="Tab Three">TabPanel Three Content</Tabs.Panel>
+ *   <Tabs.Panel id="tab-4" label="Tab Four">TabPanel Four Content</Tabs.Panel>
+ * </Tabs>
+ */
+const Tabs: FunctionComponent<PropsTabs> & StaticComponents = ({ children, id, selectedId, ariaLabel, onTabChange }) => {
+  const defaultSelectedTabId = selectedId || children[0].props.id
+  const [selectedTabId, setSelectedTabId] = useState(defaultSelectedTabId);
+
+  const handleTabChange = (newSelectedId: string) => {
+    setSelectedTabId(newSelectedId)
+
+    if (onTabChange) {
+      onTabChange(newSelectedId);
+    }
+  }
+
+  return (
+    <Tabs.Container id={id} ariaLabel={ariaLabel}>
+      <Tabs.List> 
+        {children.map(({props: { label, id }}) => (
+          <Tabs.Tab 
+            id={id + '-tab'}
+            key={id + '-tab'}
+            ariaControls={id}
+            selected={id === selectedTabId}
+            onClick={() => handleTabChange(id)}
+          >
+            {label}
+          </Tabs.Tab>
+        ))}
+      </Tabs.List>
+      <Tabs.PanelContainer>
+        {children.map(({ props: { label, id, children: tabPabelChildren } }) => (
```

Comment:
consider mapping through children only once building an array of tuples (or whatever DS) of these components to render here and above on 110.